addons/sale_coupon/models/sale_order.py

```diff
@@ -345,30 +345,36 @@ def _get_applicable_programs(self):
         self.ensure_one()
         programs = self.env['sale.coupon.program'].with_context(
             no_outdated_coupons=True
-        ).search([
-            ('company_id', 'in', [self.company_id.id, False]),
-            '|', ('rule_date_from', '=', False), ('rule_date_from', '<=', fields.Datetime.now()),
-            '|', ('rule_date_to', '=', False), ('rule_date_to', '>=', fields.Datetime.now()),
-        ], order="id")._filter_programs_from_common_rules(self)
+        ).search(self._get_applicable_programs_domain(), order="id")._filter_programs_from_common_rules(self)
```

Comment:
In the description of merge request i try to said thath is best put a method that it do the domain instead of
 a big domain in a line. It is not flexible